### PR TITLE
Transform about centre

### PR DESCRIPTION
**Pull request by nontas**
This PR adds a general function for transforming about an image's centre. In general, this probably only makes sense for Affine transforms - though any transform is supported via the transform chain.

It also adds a new 2D shear transform. The shear transform has the following formulation:

```
array([[1,        tan(phi), 0], 
       [tan(psi), 1,        0],
       [0,        0,        1]])
```

This can be useful for data perturbation (distortions).

#### Example
The following code:
```
%matplotlib inline
import matplotlib.pyplot as plt
import menpo.io as mio
from menpo.transform import Affine, Rotation, shear_about_centre

im = mio.import_builtin_asset.lenna_png()
shear_tr = Affine.init_from_2d_shear(25, 10)
shear_tr = Rotation.init_from_2d_ccw_angle(45)

plt.subplot(141)
im.view_landmarks(render_axes=True)
plt.title('Original')

plt.subplot(142)
im.transform_about_centre(shear_tr).view_landmarks(render_axes=True)
plt.title('Sheared - No Retain Shape')

plt.subplot(143)
im.transform_about_centre(shear_tr, retain_shape=True).view_landmarks(render_axes=True)
plt.title('Sheared - Retain Shape')

plt.subplot(144)
bbox = im.landmarks['LJSON'].lms.bounding_box()
bbox.view(line_width=2, line_colour='b', render_markers=False)
shear_about_centre(bbox, 25, 10).apply(bbox).view(line_width=2, line_colour='r', 
                                                  render_markers=False, 
                                                  figure_size=(17, 17))
plt.title('Example')
```

generates the following result:
download

### CODE REVIEW

**Comment by patricksnape** [comment]
OK - what if we add something like `affine_around_centre` that applies some `Affine` transform around the centre and we abstract out the whole `-translation, affine, translation` step?

File: menpo/image/base.py [hunk 1]
```diff
@@ -2257,6 +2258,101 @@ def rotate_ccw_about_centre(self, theta, degrees=True, retain_shape=False,
             shape, trans.pseudoinverse(), order=order, warp_landmarks=True,
             cval=cval, return_transform=return_transform)
 
+    def shear_about_centre(self, phi, psi, degrees=True, retain_shape=False,
```
Comment (patricksnape) [inline menpo/image/base.py]: I think this is better (more general) as `transform_about_centre` where we just make sure that we apply the translation and then the translation back?

File: menpo/transform/compositions.py [hunk 2]
```diff
@@ -60,3 +62,75 @@ def rotate_ccw_about_centre(obj, theta, degrees=True):
...
+    Raises
+    ------
+    ValueError
+        Shearing can only be applied on 2D objects
+    """
+    if obj.n_dims != 2:
+        raise ValueError('Shearing can only be applied on 2D objects')
+    # Create shearing and translation transforms
+    a = create_2d_shear_transform(phi, psi, degrees=degrees)
+    t = Translation(-obj.centre(), skip_checks=True)
```
Comment (patricksnape) [inline menpo/transform/compositions.py]: I think we should abstract this logic out.

**Comment by patricksnape** [comment]
I have refactored the PR for a more general transform about centre - maintaining the shear logic.

**Comment by nontas** [comment]
Good change @patricksnape ! I tried this and it works. I simply added two usage examples in the `Image.transform_about_centre()` doc. 
+1

**Pull request merged**